src/payment/strategies/square/square-payment-strategy.ts

```diff
@@ -103,27 +130,108 @@ export default class SquarePaymentStrategy extends PaymentStrategy {
                     }
                 },
+
                 unsupportedBrowserDetected: () => {
                     deferred.reject(new UnsupportedBrowserError());
                 },
-                cardNonceResponseReceived: (errors, nonce) => {
-                    this._cardNonceResponseReceived(errors, nonce);
+
+                cardNonceResponseReceived: (errors?, nonce?, cardData?, billingContact?, shippingContact?) => {
+
+                    if (cardData && cardData.digital_wallet_type !== DigitalWalletType.none) {
+                        this._setExternalCheckoutData(nonce, cardData, billingContact, shippingContact)
+                            .then(() => this._paymentInstrumentSelected(methodId))
+                            .then(() => squareOptions.onPaymentSelect && squareOptions.onPaymentSelect());
+                    } else {
+                        this._cardNonceResponseReceived(nonce, errors);
+                    }
+                },
+
+                methodsSupported: (methods: any) => {
+                    const masterpassBtn = document.getElementById('sq-masterpass');
+                    if (masterpassBtn !== null) {
+                        if (methods.masterpass === true) {
+                            masterpassBtn.style.display = 'inline-block';
+                        }
+                    }
+                },
+
+                /*
+                 * callback function: createPaymentRequest
+                 * Triggered when: a digital wallet payment button is clicked.
+                */
+                createPaymentRequest: () => {
+                    const state = this._store.getState();
+                    const checkout = state.checkout.getCheckout();
+                    const storeConfig = state.config.getStoreConfig();
+
+                    if (!checkout) {
+                        throw new MissingDataError(MissingDataErrorType.MissingCheckout);
+                    }
+
+                    if (!storeConfig) {
+                        throw new MissingDataError(MissingDataErrorType.MissingCheckoutConfig);
+                    }
+
+                    return {
+                        requestShippingAddress: true,
+                        requestBillingInfo: true,
+                        currencyCode: storeConfig.currency.code,
+                        countryCode: 'US',
+                        total: {
+                            label: storeConfig.storeProfile.storeName,
+                            amount: checkout.subtotal.toString(),
+                            pending: false,
+                        },
+                    };
                 },
             },
         };
     }
 
-    private _cardNonceResponseReceived(errors: any, nonce: string): void {
+    private _paymentInstrumentSelected(methodId: string) {
+        return this._store.dispatch(this._paymentStrategyActionCreator.widgetInteraction(() => {
```

Comment:
Just had a second look. You need to fire the widget interaction event as soon as we begin the nonce retrieval process. This is so that developers can check `CheckoutStoreStatusSelector#isPaymentStepPending` value to determine whether it is possible to submit payment. If the payment step is still pending (i.e.: things are happening in the background), they can use this flag to disable / enable the submit button. I can see that you have another asynchronous process (`_setExternalCheckoutData`) before this so you need to wrap that also.